File: kotlin-advent-of-code/src/main/kotlin/advent/twenty_eighteen/Day01ChronalCalibration.kt
package advent.twenty_eighteen

import advent.common.DailyProblem
import advent.twenty_seventeen.Day01InverseCaptcha
import java.util.function.IntFunction
import java.util.stream.Collectors

class Day01ChronalCalibration : DailyProblem<Int, Int> {
    override var part1Answer: Int = 0
    override var part2Answer: Int = 0

    constructor(filename: String) {
        val changes = Day01InverseCaptcha::class.java.getResource(filename).openStream().bufferedReader().lines()
            .map { line -> Integer.parseInt(line) }.collect(Collectors.toList())

        var frequency = 0
        for (change in changes) {
            frequency += change
        }
        part1Answer = frequency

        frequency = 0
        var frequencies = HashSet<Int>()
        var idx = 0
        while (!frequencies.contains(frequency)) {
            if (idx >= changes.size) {
                idx = 0
            }
            frequencies.add(frequency)
            frequency += changes[idx]
            idx++
        }
        part2Answer = frequency
    }
}

File: dotnet-advent-of-code/Aoc2018/Day01ChronalCalibration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Common;

namespace TwentyEighteen
{
    public class Day01ChronalCalibration : IDailyProblem<int, int>
    {
        public int Part1Answer { get; }
        public int Part2Answer { get; }

        public Day01ChronalCalibration(String filename)
        {
            List<int> frequencyChanges = InputFileReader.ReadLines(filename, Convertors.StringToInt);

            Part1Answer = frequencyChanges.Sum();

            var seen = new HashSet<int>();
            var frequency = 0;
            var index = 0;
            while (!seen.Contains(frequency))
            {
                if (index >= frequencyChanges.Count)
                {
                    index = 0;
                }

                seen.Add(frequency);
                frequency = frequency + frequencyChanges[index];
                index++;
            }

            Part2Answer = frequency;
        }
    }
}